src/renderers/dom/client/ReactMount.js

```diff
@@ -724,6 +729,15 @@ var ReactMount = {
       }
     }
   },
+  /**
+   * Rather than count mounts and unmounts, we query the dom for the DOMProperty.ROOT_ATTRIBUTE_NAME. This is used to
+   * dispose of global event listeners, when the last component is removed/unmounted.
+   * @returns Returns the number of react tagged root elements.
+   * @private
+   */
+  _countReactRootElements: function() {
+    return document.querySelectorAll('[ROOT_ATTR_NAME]').length;
+  },
```

Comment:
???
This is the *only* big q, does document.querySelectAll exist in all currently supported react browsers, or do i need a fallback etc.